### PR TITLE
Update hover-text-and-formatting.md

### PR DESCRIPTION
**Pull request by rl-utility-man**
1)  added an example that formats the elements of custom_data in plotly express
2)  fixed a bug in the "advanced hovertemplate" example which caused it to display the square root of population.   Modified that example to follow the main dataframe is called "df" convention.  Added a country name field to that example and comments about the usage of key parameters.
3)  I did not remove the "Adding other data to the hover with customdata and a hovertemplate" example, but I believe it is now redundant and that the revised ### Advanced Hover Template example is more compliant with expectations like using meaningful data over random data.  If you removed it, it would make sense to rename "Advanced Hover Template" to include the phrase:  "Adding other data to the hover with customdata and a hovertemplate"

### Documentation PR

- [ X] I've [seen the `doc/README.md` file](https://github.com/plotly/plotly.py/blob/master/doc/README.md)
- [ yes, 5.20!] This change runs in the current version of Plotly on PyPI and targets the `doc-prod` branch OR it targets the `master` branch
- [ X] If this PR modifies the first example in a page or adds a new one, it is a `px` example if at all possible
- [ X] Every new/modified example has a descriptive title and motivating sentence or paragraph
- [ X] Every new/modified example is independently runnable
- [ X ] Every new/modified example is optimized for short line count	and focuses on the Plotly/visualization-related aspects of the example rather than the computation required to produce the data being visualized
- [ X] Meaningful/relatable datasets are used for all new examples instead of randomly-generated data where possible
- [N/A ] The random seed is set if using randomly-generated data in new/modified examples
- [ N/A] New/modified remote datasets are loaded from https://plotly.github.io/datasets and added to https://github.com/plotly/datasets
- [ X] Large computations are avoided in the new/modified examples in favour of loading remote datasets that represent the output of such computations
- [ X] Imports are `plotly.graph_objects as go` / `plotly.express as px` / `plotly.io as pio`
- [ X] Data frames are always called `df`
- [ X ] `fig = <something>` call is high up in each new/modified example (either `px.<something>` or `make_subplots` or `go.Figure`)
- [ X] Liberal use is made of `fig.add_*` and `fig.update_*` rather than `go.Figure(data=..., layout=...)` in every new/modified example
- [ ] Specific adders and updaters like `fig.add_shape` and `fig.update_xaxes` are used instead of big `fig.update_layout` calls in every new/modified example
- [X ] `fig.show()` is at the end of each new/modified example
- [ X] `plotly.plot()` and `plotly.iplot()` are not used in any new/modified example
- [ X] Hex codes for colors are not used in any new/modified example in favour of [these nice ones](https://github.com/plotly/plotly.py/issues/2192)

### CODE REVIEW

File: doc/python/hover-text-and-formatting.md [hunk 1]
```diff
@@ -252,6 +252,58 @@ print("user_defined hovertemplate:", fig.data[0].hovertemplate)
...
+
+df.rename(columns={"gdpPercap":'GDP per capita', "lifeExp":'Life Expectancy (years)'}, inplace=True)
+
+fig=px.scatter(df, 
+               x='GDP per capita',
+               y='Life Expectancy (years)', 
+               color='continent', 
+               size=np.sqrt(df['pop']),           
+               # Specifying data to make availabe to the hovertemplate
+               # The px custom_data parameter has an underscore, whike the analogous graph objects customdata parameter has no underscore.
```
Comment (Coding-with-Adam) [inline doc/python/hover-text-and-formatting.md:279]:
```suggestion
               # The px custom_data parameter has an underscore, while the analogous graph objects customdata parameter has no underscore.
```

File: doc/python/hover-text-and-formatting.md [hunk 2]
```diff
@@ -252,6 +252,58 @@ print("user_defined hovertemplate:", fig.data[0].hovertemplate)
...
+               # The px custom_data parameter is a list of column names in the data frame, while the graph objects customdata parameter expects a data frame or a numpy array.
+               custom_data=['country', 'continent', 'pop'], 
+)
+
+# Plotly express does not have a hovertemplate paramter in the graph creation function, so we apply the template with update_traces
```
Comment (Coding-with-Adam) [inline doc/python/hover-text-and-formatting.md:284]:
```suggestion
# Plotly express does not have a hovertemplate parameter in the graph creation function, so we apply the template with update_traces
```

**Review by Coding-with-Adam — APPROVED** [review]
nice catch with the square root population displaying. 
This new code example is much better, @rl-utility-man . Thank you. 

Although redundant, I think it's ok to keep the section on "Adding other data to the hover with customdata and a hovertemplate".  Having an example with a different graph type can be helpful to some.

**Comment by Coding-with-Adam** [comment]
@LiamConnors are you ok with these suggested commits?

**Comment by rl-utility-man** [comment]
@LiamConnors is there anything I can do to help finish this up and get it merged?  Thanks!

File: doc/python/hover-text-and-formatting.md [hunk 3]
```diff
@@ -252,6 +252,58 @@ print("user_defined hovertemplate:", fig.data[0].hovertemplate)
 fig.show()
 ```
 
+### Specifying the formatting and labeling of custom fields in a Plotly Express figure using a hovertemplate
+
+This example adds custom fields to a Plotly Express figure using the custom_data parameter and then  adds a hover template that applies d3 formats to each element of the customdata[n] array and uses HTML to customize the fonts and spacing.
```
Comment (LiamConnors) [inline doc/python/hover-text-and-formatting.md:257]:
```suggestion
This example adds custom fields to a Plotly Express figure using the custom_data parameter and then adds a hover template that applies d3 formats to each element of the customdata[n] array and uses HTML to customize the fonts and spacing. 
```

File: doc/python/hover-text-and-formatting.md [hunk 4]
```diff
@@ -312,21 +363,31 @@ continent_data = {continent:df_2007.query("continent == '%s'" %continent)
...
+            y=df['lifeExp'],
+            marker_size=df['size'],
+            name=continent_name,
+
+            # The next three parameters specify the hover text
+            # Text supports just one customized field per trace, but is simple to implement
+            text=df['continent'],  
+            # Custom data supports multiple fields through numeric indices in the hovertemplate 
+            # If I were not looking for an opportunity to demonstrate the text parameter, 
+            # I would likely just add continent as a third customdata field.
```
Comment (LiamConnors) [inline doc/python/hover-text-and-formatting.md:379]:
```suggestion
            # Custom data supports multiple fields through numeric indices in the hovertemplate 
            # In we weren't using the text parameter in our example, 
            # we could instead add continent as a third customdata field.
```

@rl-utility-man just want to keep that language similar to the language earlier in the example. Does this still read as you intended?

File: doc/python/hover-text-and-formatting.md [hunk 5]
```diff
@@ -312,21 +363,31 @@ continent_data = {continent:df_2007.query("continent == '%s'" %continent)
...
+    fig.add_trace(
+        go.Scatter(
+            x=df['gdpPercap'],
+            y=df['lifeExp'],
+            marker_size=df['size'],
+            name=continent_name,
+
+            # The next three parameters specify the hover text
+            # Text supports just one customized field per trace, but is simple to implement
+            text=df['continent'],
```
Comment (LiamConnors) [inline doc/python/hover-text-and-formatting.md:376]:
```suggestion
            # The next three parameters specify the hover text
            # Text supports just one customized field per trace 
            # and is implemented here with text=df['continent'],  
```

File: doc/python/hover-text-and-formatting.md (hunk 4 above)
Comment (rl-utility-man) [reply doc/python/hover-text-and-formatting.md:379]: Looks great!

File: doc/python/hover-text-and-formatting.md (hunk 5 above)
Comment (rl-utility-man) [reply doc/python/hover-text-and-formatting.md:376]: Looks fine!  Many thanks!

File: doc/python/hover-text-and-formatting.md [hunk 6]
```diff
@@ -283,6 +283,58 @@ print("user_defined hovertemplate:", fig.data[0].hovertemplate)
 fig.show()
 ```
 
+### Specifying the formatting and labeling of custom fields in a Plotly Express figure using a hovertemplate
+
+This example adds custom fields to a Plotly Express figure using the custom_data parameter and then adds a hover template that applies d3 formats to each element of the customdata[n] array and uses HTML to customize the fonts and spacing. 
+
+```
```
Comment (LiamConnors) [inline doc/python/hover-text-and-formatting.md:290]:
```suggestion
```python
```

File: doc/python/hover-text-and-formatting.md [hunk 7]
```diff
@@ -283,6 +283,58 @@ print("user_defined hovertemplate:", fig.data[0].hovertemplate)
...
+               custom_data=['country', 'continent', 'pop'], 
+)
+
+# Plotly express does not have a hovertemplate parameter in the graph creation function, so we apply the template with update_traces
+fig.update_traces(
+    hovertemplate = 
+                "<b>%{customdata[0]}</b><br>" +
+                "<b>%{customdata[1]}</b><br><br>" +
+                "GDP per Capita: %{x:$,.0f}<br>" +
+                "Life Expectation: %{y:.0f}<br>" +
```
Comment (LiamConnors) [inline doc/python/hover-text-and-formatting.md:321]:
```suggestion
                "Life Expectancy: %{y:.0f}<br>" +
```

**Pull request merged**